src/controllers/Talk_commentsController.php

```diff
@@ -130,4 +130,44 @@ public function moderateReportedComment($request, $db)
         header("Location: $uri", true, 204);
         exit;
     }
+
+    public function updateComment($request, $db)
+    {
+        $request->user_id = 14;
+
+        // must be logged in
+        if (!isset($request->user_id) || empty($request->user_id)) {
+            throw new Exception('You must log in to edit a comment', 401);
+        }
+
+        $new_comment_body = $request->getParameter('comment');
+        if (empty($new_comment_body)) {
+            throw new Exception('The field "comment" is required', 400);
+        }
+
+        $comment_id = $this->getItemId($request);
+        $comment_mapper = new TalkCommentMapper($db, $request);
+        $comment = $comment_mapper->getRawComment($comment_id);
+
+        if (false === $comment) {
+            throw new Exception('Comment not found', 404);
+        }
+
+        if ($comment['user_id'] != $request->user_id) {
+            throw new Exception('You are not the comment author', 401);
```

Comment:
This should be a 403 as return code (HTTP-Status Forbidden) as we know the user but that user just isn't allowed doing this action.